src/coord/src/coord.rs

```diff
@@ -2143,7 +2153,11 @@ impl Coordinator {
                 kind: catalog::ErrorKind::ItemAlreadyExists(_),
                 ..
             })) if if_not_exists => Ok(ExecuteResponse::CreatedSecret { existed: true }),
-            Err(err) => Err(err),
+            Err(err) => {
+                // best-effort attempt to drop the newly created secret, ignore any failures
+                let _ = self.secrets_controller.apply(vec![SecretOp::Delete { id }]);
```

Comment:
This logic should go after the call to catalog_transact, which will also handle the case that the catalog can't do an operation.

Reply:
this is as specified by @benesch in https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20220303_secrets.md#transaction-integration. The idea is to error on the side of orphaned files on disk, in case the transaction gets interrupted.

Reply:
Ah we're saying the same thing, I just misread this. I thought this was in the closure, but it is indeed handling the return of catalog_transact, which is what I was saying. Yeah, this is right.